### ISSUE TITLE
npm config `map` is not being set properly

### ISSUE THREAD

**Issue by BigAB** [stealjs/steal#1208]
__How often can you reproduce it?__

- [x] Always
- [ ] Sometimes
- [ ] Rarely
- [ ] Unable
- [ ] I didn’t try

__Description:__
Steal config `map` does not seem to be being set properly.

__Steps to reproduce:__

> 1. Include a link to your project (if possible)  

Here is a gist you can clone to show the issue.
https://gist.github.com/BigAB/ae311de12272af411969f477e6f8c1a5

`preact-compat` should be a drop in replacement for `react` and `react-dom` and so map should swap them out fine. But in the gist, if you serve it you can see steal is still trying to import `react`.

The relevant part of the gist is mostly this in `package.json:
```
  "steal": {
    "map": {
      "react": "preact-compat",
      "react-dom": "preact-compat"
    }
  }
```

> 2. Detail the exact steps taken to produce the problem  

Clone the gist. Run `npm install`. Serve the root. See in the console that react is trying to load and `System.map` only has `@loader/@laoder` and `@steal/@steal` keys.

<img width="1002" alt="screen shot 2017-06-05 at 9 45 50 am" src="https://cloud.githubusercontent.com/assets/76821/26791288/d4c9e106-49d3-11e7-942a-4baae3c6c6d9.png">

> 3. Include a gif if applicable

![steal-config-map-issue](https://cloud.githubusercontent.com/assets/76821/26791942/f8d5de7c-49d5-11e7-98af-31dbf71250aa.gif)

__Expected results:__

I expect everything to work, for every call to `import React from 'react` would return the `preact-compat` module instead

__Actual results:__
Steal fails to load `react` because it isn't there, and the config in `npm` does not even appear to be merged into `System.map`

__Environment:__

| Software                     | Version
| --------------------- | -------
| Steal version              | 1.5.2
| Steal-tools version    | 1.3.5
| node -v                       | v8.0.0
| npm -v                        | 5.0.2
| Browser                      | Chrome 58.0.3029.110
| Operating system      | macOS Sierra 10.12.5

**Comment by matthewp** [comment]
This is caused by progressive fetching of package.jsons. The problem is that the left-hand side is not an npm module, so the code that is supposed to load preact-compat is never triggered.  I suppose the fix is that we should fetch the right-hand side packages right away and not wait, if the left-hand side isn't npm.  Going to look into what this would take to fix.

**Comment by matthewp** [comment]
Blocked by https://github.com/stealjs/steal/issues/1210

**Comment by matthewp** [comment]
Breaking test added in ff9c28d7e533fc661e165e6dac09c2207b4c92ea
